xbmc/utils/LegacyPathTranslation.cpp

```diff
@@ -11,95 +11,91 @@
+    {"videodb://3/4", "videodb://musicvideos/artists"},
+    {"videodb://3/5", "videodb://musicvideos/albums"},
+    {"videodb://3/9", "videodb://musicvideos/tags"},
+    {"videodb://3", "videodb://musicvideos"},
+    {"videodb://4", "videodb://recentlyaddedmovies"},
+    {"videodb://5", "videodb://recentlyaddedepisodes"},
+    {"videodb://6", "videodb://recentlyaddedmusicvideos"}};
 
 // ATTENTION: Make sure the longer match strings go first
 // because the string match is performed with StringUtils::StartsWith()
-static Translator s_musicDbTranslator[] = {
-  { "musicdb://10",   "musicdb://singles" },
-  { "musicdb://1",    "musicdb://genres" },
-  { "musicdb://2",    "musicdb://artists" },
-  { "musicdb://3",    "musicdb://albums" },
-  { "musicdb://4",    "musicdb://songs" },
-  { "musicdb://5/1",  "musicdb://top100/albums" },
-  { "musicdb://5/2",  "musicdb://top100/songs" },
-  { "musicdb://5",    "musicdb://top100" },
-  { "musicdb://6",    "musicdb://recentlyaddedalbums" },
-  { "musicdb://7",    "musicdb://recentlyplayedalbums" },
-  { "musicdb://8",    "musicdb://compilations" },
-  { "musicdb://9",    "musicdb://years" }
-};
+const auto s_musicDbTranslator =
+    std::unordered_map<std::string, std::string>{{"musicdb://10", "musicdb://singles"},
+                                                 {"musicdb://1", "musicdb://genres"},
+                                                 {"musicdb://2", "musicdb://artists"},
+                                                 {"musicdb://3", "musicdb://albums"},
+                                                 {"musicdb://4", "musicdb://songs"},
+                                                 {"musicdb://5/1", "musicdb://top100/albums"},
+                                                 {"musicdb://5/2", "musicdb://top100/songs"},
+                                                 {"musicdb://5", "musicdb://top100"},
+                                                 {"musicdb://6", "musicdb://recentlyaddedalbums"},
+                                                 {"musicdb://7", "musicdb://recentlyplayedalbums"},
+                                                 {"musicdb://8", "musicdb://compilations"},
+                                                 {"musicdb://9", "musicdb://years"}};
 
-#define MusicDbTranslatorSize sizeof(s_musicDbTranslator) / sizeof(Translator)
-
-std::string CLegacyPathTranslation::TranslateVideoDbPath(const CURL &legacyPath)
+std::string TranslatePath(const std::string& legacyPath,
+                          const std::unordered_map<std::string, std::string>& translationMap)
 {
-  return TranslatePath(legacyPath.Get(), s_videoDbTranslator, VideoDbTranslatorSize);
+  std::string newPath = legacyPath;
+  const std::string lowPath = StringUtils::ToLower(legacyPath);
+  const auto it = std::find_if(translationMap.begin(), translationMap.end(),
+                               [&lowPath](const auto& e) { return lowPath.starts_with(e.first); });
```

Comment:
the old way was using StartsWithNoCase it was converting to lowercase both values,
now you are comparing one lower string, and you assume that the other value is lowered, but imo not so correct, an oversight can cause problems

on "translator" lists above there are comment "ATTENTION:...." that refer to StartsWithNoCase but if you change method all comments should be updated

Reply:
we control the static table of values. this is entirely on purpose. instead of lower-casing both strings for every comparison, lowercase the user provided data that we do not control once. it is an optimization.

comments updated.